be/src/vec/functions/function_map.cpp

```diff
@@ -440,13 +491,409 @@ class FunctionStrToMap : public IFunction {
+        const IColumn* map_values_column = &map_values_nullable.get_nested_column();
+        const auto& map_values_nullmap =
+                map_values_nullable.get_null_map_column().get_data().data();
+
+        auto result_column = ColumnUInt8::create(map_offsets.size(), 0);
+        auto& result_data = result_column->get_data();
+
+        const UInt8* map_row_nullmap = nullptr;
+        if (map_row_nullmap_col) {
+            map_row_nullmap =
+                    assert_cast<const ColumnUInt8&>(*map_row_nullmap_col).get_data().data();
+        }
+
+        auto matches = ColumnUInt8::create(map_keys_column->size(), 1);
+
+        // matches &= key_compare
+        _dispatch_column_comparison(key_type, *map_keys_column, map_keys_nullmap, key_column,
+                                    key_nullmap, map_offsets, map_row_nullmap, key_is_const,
+                                    *matches);
+
+        // matches &= value_compare
+        _dispatch_column_comparison(value_type, *map_values_column, map_values_nullmap,
+                                    value_column, value_nullmap, map_offsets, map_row_nullmap,
+                                    value_is_const, *matches);
+
+        // aggregate results by map boundaries
+        auto& matches_data = matches->get_data();
+        for (size_t row = 0; row < map_offsets.size(); ++row) {
+            if (map_row_nullmap && map_row_nullmap[row]) {
+                // result is null for this row
+                continue;
+            }
+
+            size_t map_start = row == 0 ? 0 : map_offsets[row - 1];
+            size_t map_end = map_offsets[row];
+
+            bool found = false;
+            for (size_t i = map_start; i < map_end && !found; ++i) {
+                if (matches_data[i]) {
+                    found = true;
+                    break;
+                }
+            }
+            result_data[row] = found;
+        }
+
+        if (map_row_nullmap_col) {
+            return ColumnNullable::create(std::move(result_column), map_row_nullmap_col);
+        }
+        return result_column;
+    }
+
+    // type comparability check and dispatch
+    Status _execute_type_check_and_dispatch(Block& block, const ColumnNumbers& arguments,
+                                            uint32_t result) const {
+        // get type information
+        auto map_type = remove_nullable(block.get_by_position(arguments[0]).type);
+        const auto* map_datatype = assert_cast<const DataTypeMap*>(map_type.get());
+        auto map_key_type = remove_nullable(map_datatype->get_key_type());
+        auto map_value_type = remove_nullable(map_datatype->get_value_type());
+        auto search_key_type = remove_nullable(block.get_by_position(arguments[1]).type);
+        auto search_value_type = remove_nullable(block.get_by_position(arguments[2]).type);
+
+        bool key_types_comparable = type_comparable(map_key_type->get_primitive_type(),
```

Comment:
In FE we use `AnyDataType` with same slot or `FollowToAnyDataType`. they both will inference type to same. so I think `map_key_type->get_primitive_type()` and `search_key_type->get_primitive_type()` here should be same type?